src/test/java/pages/HomePage.java

```diff
@@ -1,13 +1,16 @@
 package pages;
 
 import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 
 public class HomePage extends BasePage {
 
     By userAvatarIcon = By.cssSelector("img.avatar");
-
+    By firstPlaylist = By.cssSelector(".playlist:nth-child(3)");
+    By playlistInputField = By.cssSelector("input[name='name']");
```

Comment:
You should use encapsulation of non-static and non-final fields.

If a variable is used only by other methods of that class, it should be declared as private.

I recommend:  make all the locators and WebElements private, because private keyword in java allows most restrictive access to variables and methods and offer strongest form of Encapsulation. 

For example

    ```
    private By shuffleBtnLocator = By.cssSelector(".btn-shuffle-all");
    private By firstSongLocator = By.cssSelector(".play");

```

Reply:
We haven't learn about encapsulation of non-static and non-final fields. So I don't know what does it do. Presentation slides for lesson 23 have the code for HW 22 and they don't use it.

Reply:
**Encapsulation** is a way to hide the inner workings of an object and only allow access through specific methods. 

**Non-static and non-final fields** are instance variables of an object, and encapsulating them involves making them private and providing public methods to access and modify their values. This helps improve data integrity and maintainability.

Sorry, I wanted you to learn even more (

Reply:
Thanks for trying! But it's still hard for me to understand this concept.